app/src/main/java/org/kiwix/kiwixmobile/webserver/WebServerHelper.java

```diff
@@ -0,0 +1,116 @@
+package org.kiwix.kiwixmobile.webserver;
+
+import android.util.Log;
+import androidx.annotation.NonNull;
+import java.net.InetAddress;
+import java.net.NetworkInterface;
+import java.net.SocketException;
+import java.util.ArrayList;
+import java.util.Enumeration;
+import org.kiwix.kiwixlib.JNIKiwixException;
+import org.kiwix.kiwixlib.JNIKiwixLibrary;
+import org.kiwix.kiwixlib.JNIKiwixServer;
+
+/**
+ * WebServerHelper class is used to set up the suitable environment i.e. getting the
+ * ip address and port no. before starting the WebServer
+ * Created by Adeel Zafar on 18/07/2019.
+ */
+
+public class WebServerHelper {
+  public static boolean isServerStarted;
+  private static int port;
+  private final JNIKiwixLibrary kiwixLibrary = new JNIKiwixLibrary();
+  private final JNIKiwixServer kiwixServer = new JNIKiwixServer(kiwixLibrary);
+  private static final String TAG = "WebServerHelper";
+
+  public WebServerHelper() {
+
+  }
+
+  public boolean startServerHelper(@NonNull ServerStateListener stateListener,
```

Comment:
we don't want to pass around the StateListener. You may want this class to have its own callbacks

Reply:
Fixed in 47d11384fd390d63325cdce794bd88c75dd3c1f3